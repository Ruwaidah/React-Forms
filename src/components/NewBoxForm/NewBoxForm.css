.NewBoxForm {
    border-bottom: black solid 1px;
    padding-bottom: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: rgb(251, 214, 175);
}

.NewBoxForm form {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 40%;
    font-weight: bold;

} 

.NewBoxForm form input[type="submit"] {
    font-family: 'Edu SA Beginner', cursive;
    font-weight: bold;
    width: 60px;

}