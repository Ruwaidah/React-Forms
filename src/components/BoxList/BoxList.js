import { useState } from "react";
import Box from "../Box/Box";
import NewBoxForm from "../NewBoxForm/NewBoxForm";
import "./BoxList.css";
const BoxList = () => {
  const [boxs, setBoxs] = useState([]);
  const addBox = (newBox) => {
    setBoxs([...boxs, newBox]);
  };

  const deleteBox = (i) => setBoxs(boxs.filter((box, index) => index !== i));
  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      <div className="BoxList-div">
        {boxs.map((box, idx) => (
          <Box box={box} key={idx} idx={idx} deleteBox={deleteBox} />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
