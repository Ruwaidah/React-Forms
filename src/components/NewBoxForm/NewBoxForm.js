import { useState } from "react";
import "./NewBoxForm.css";

const NewBoxForm = (props) => {
  const initial = {
    width: "",
    height: "",
    color: "#000000",
  };
  const [formData, setFormData] = useState(initial);

  const onChange = (e) =>
    setFormData({ ...formData, [e.target.name]: e.target.value });

  const onSubmit = (e) => {
    e.preventDefault();
    props.addBox(formData);
    setFormData(initial);
  };
  return (
    <div className="NewBoxForm">
      <h3>Create New Box</h3>
      <form onSubmit={onSubmit}>
        <label htmlFor="width">
          Width:
          <input
            id="width"
            type="number"
            max={300}
            min={20}
            name="width"
            placeholder="width"
            required
            value={formData.width}
            onChange={onChange}
          />
        </label>
        <label htmlFor="height">
          Height:
          <input
            id="height"
            type="number"
            max={300}
            min={20}
            name="height"
            value={formData.height}
            placeholder="height"
            required
            onChange={onChange}
          />
        </label>
        <label htmlFor="color">
          Color:
          <input
            id="color"
            value={formData.color}
            name="color"
            type="color"
            onChange={onChange}
          />
        </label>
        <input type="submit" value="Submit" />
      </form>
    </div>
  );
};

export default NewBoxForm;
