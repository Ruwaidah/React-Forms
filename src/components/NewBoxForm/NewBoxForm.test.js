import { fireEvent, render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";
import BoxList from "../BoxList/BoxList";

it("renders without crashing", () => {
  render(<NewBoxForm />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", () => {
  const { queryByText, getByLabelText, qu } = render(<BoxList />);
  const inputWidth = getByLabelText("Width:");
  const inputHeight = getByLabelText("Height:");
  const inputColor = getByLabelText("Color:");
  const btn = queryByText("Submit");
  let deleteBtn = queryByText("Delete");
  expect(deleteBtn).not.toBeInTheDocument();
  fireEvent.change(inputWidth, { target: { value: "200" } });
  fireEvent.change(inputColor, { target: { value: "#7d3f3f" } });
  fireEvent.change(inputHeight, { target: { value: "200" } });
  fireEvent.click(btn);
  deleteBtn = queryByText("X");
  expect(deleteBtn).toBeInTheDocument();
});
