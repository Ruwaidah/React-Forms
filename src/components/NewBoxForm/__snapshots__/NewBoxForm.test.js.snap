// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewBoxForm"
  >
    <h3>
      Create New Box
    </h3>
    <form>
      <label
        for="width"
      >
        Width:
        <input
          id="width"
          max="300"
          min="20"
          name="width"
          placeholder="width"
          required=""
          type="number"
          value=""
        />
      </label>
      <label
        for="height"
      >
        Height:
        <input
          id="height"
          max="300"
          min="20"
          name="height"
          placeholder="height"
          required=""
          type="number"
          value=""
        />
      </label>
      <label
        for="color"
      >
        Color:
        <input
          id="color"
          name="color"
          type="color"
          value="#000000"
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
